### ISSUE TITLE
`RI` unconditionally whines about obscure path

### ISSUE THREAD

**Issue by mih** [datalad/datalad#6872]
The tests use obscure path names for good reasons. Given that, I do not understand this behavior:

```py
>>> from datalad.support.network import RI
>>> RI("/tmp/datalad-fs-DAkJS/datalad_temp_tree_test_create_subdataset_hierarchy_from_topkfwtympw/origin/ds-;&%b5{}'ΔЙקم๗あ.datc")
[WARNING] ParseResults contains params "&%b5{}'ΔЙקم๗あ.datc", which will be ignored 
PathRI(path="/tmp/datalad-fs-DAkJS/datalad_temp_tree_test_create_subdataset_hierarchy_from_topkfwtympw/origin/ds-;&%b5{}'ΔЙקم๗あ.datc")
```

The path shown above is generated by our own test helpers. `RI` seems to be cabable of handling it correctly. But in the process `URL._pr_to_fields()` is whining loudly, and inappropriately in this context.

This is both a UX issue (users have to endure this with any repository that has `&` in the directory name), and a developer X issue (I spent a considerable amount of timing hunting this message down, assuming that it is related to a bug I am investigation (given the WARNING level), but turns out to be inconsequential).

**Comment by adswa** [comment]
It seems that `urlparse` follows this [standard](https://datatracker.ietf.org/doc/html/rfc1808.html) URL scheme with some special characters that are present in that obscure path and confuse it (in this particular case, its ``;``) :

> The URL syntax is dependent upon the scheme.  Some schemes use reserved characters like "?" and ";" to indicate special Components, while others just consider them to be part of the path.  However,  there is enough uniformity in the use of URLs to allow a parser to resolve relative URLs based upon a single, generic-RL syntax.  This generic-RL syntax consists of six components:

    <scheme>://<net_loc>/<path>;<params>?<query>#<fragment>

And a valid usecase with parameters in the URL would look like this for e.g. FTP:
   
      <cwd1>/<cwd2>/.../<cwdN>/<name>;type=<typecode>

I guess its in general helpful to have information that parameters in URLs aren't honored if you explicitly put them there. But I agree that in the context of this random obscure file path, or real-life paths that contain a ``;``, its not helpful at all. Would reducing the log level of this message to debug be a better balance?

**Comment by adswa** [comment]
This was fixed in #6891
